### PR TITLE
Fix hermesFlagsRelease not working with multiple productFlavors

### PR DESCRIPTION
**Pull request by grit96**
Ref https://github.com/facebook/react-native/issues/25601#issuecomment-510856047.

## Summary

The `hermesFlagsRelease` option only works with the release build type, but not with other build types. This PR considers other build types as release and enables `hermesFlagsRelease` to apply to any non-debug build type.

## Changelog

[Android] [Fixed] - Fix hermesFlagsRelease not working with multiple productFlavors

## Test Plan

Set `hermesFlagsRelease: ["-w"]` in `android/app/build.gradle` and ensure warnings are hidden when running `./gradlew assembleRelease` and `./gradlew assembleLive`.

### CODE REVIEW

File: react.gradle [hunk 1]
```diff
@@ -186,7 +186,7 @@ afterEvaluate {
                     def hermesFlags;
                     def hbcTempFile = file("${jsBundleFile}.hbc")
                     exec {
-                        if (targetName.toLowerCase().contains("release")) {
+                        if (!targetName.toLowerCase().contains("debug")) {
```
Comment (ShikaSD) [inline react.gradle:189]: I am not sure changing this logic for everyone won’t break the build somewhere (assuming that some people may have custom debug-oriented buildTypes). Maybe the best solution would be to introduce a lambda similar to `enableHermesForVariant` or flag like we have with `bundleIn$targetName`, and configure this logic from outside.

Comment (grit96) [reply react.gradle:189]: Yes that probably is a better solution

Comment (sota000) [reply react.gradle:189]: @grit96, were you going to make the change based on the suggestion?

**Comment by sota000** [comment]
Hi! Closing this PR since the comment hasn't been addressed for a while. Please reopen and ping me if you are still up for working on this. Thanks for your time!

**Comment by grit96** [comment]
@sota000 sorry for not getting back sooner - I've made the changes in a new PR https://github.com/facebook/react-native/pull/32281

**Pull request closed without merging**